### PR TITLE
Add Windows CI jobs

### PR DESCRIPTION
**Pull request by disktnk**
- only python 3.7 and cuda 10.0
- on cpu and gpu
- [x] need to fix `tests\chainer_tests\test_init_docstring.py` fail

### CODE REVIEW

**Comment by niboshi** [comment]
How can we test the PR?

**Comment by disktnk** [comment]
`chainer.win.{gpu,cpu}` project has already created on pfnci, so add these project to webhook and write comment like "test please", then the jobs will start. But I don't know a correct way to add webhook on this project so have not added webhook yet. CC: @kmaehashi

**Comment by niboshi** [comment]
Jenkins, test this please

File: .pfnci/install_cupy.bat [hunk 1]
```diff
@@ -0,0 +1,23 @@
+@echo off
+
+set CHAINER_LATEST_MAJOR_VER=7
```
Comment (niboshi) [inline .pfnci/install_cupy.bat]: Should we update the version number on each major release?
How about other approach?
(What I come up with immediately is to check presence of the branch `v%CHAINER_MAJOR_VER%` after clone).

File: .pfnci/install_cupy.bat [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+if %CHAINER_MAJOR_VER% equ %CHAINER_LATEST_MAJOR_VER% (
+    git clone https://github.com/cupy/cupy.git
+) else (
+    git clone -b v%CHAINER_MAJOR_VER% https://github.com/cupy/cupy.git
+)
+cd cupy
+pip install cython
+pip install -e . -vvv
+
+cd %TEST_HOME%
```
Comment (niboshi) [inline .pfnci/install_cupy.bat]: How about `pushd`/`popd`?
Or, I don't think `cd` is necessary (`pip install -e cupy/ -vvv`).

File: .pfnci/run_test.bat [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+set CUDA_VER=%1
+set PY_VER=%2
+set GPU_LIMIT=%3
+set CUDA_PATH=CUDA_PATH_V%CUDA_VER%
+set PY_PATH=C:\Development\Python\Python%PY_VER%
+set PATH=%CUDA_PATH%\bin;%CUDA_PATH%\libnvvp;%PY_PATH%;%PY_PATH%\Scripts\%PATH%
+
+set PYTEST_ATTR=not slow and not ideep
+
+curl -o xpytest.exe --insecure -L https://github.com/disktnk/xpytest/releases/download/v0.0.1/xpytest.exe
```
Comment (niboshi) [inline .pfnci/run_test.bat]: This line looks very insecure actually...

**Comment by disktnk** [comment]
thank you for your revies

>can I expect that the tests\chainer_tests\test_init_docstring.py issue will be solved in this PR?

I don't have a plan to fix it.

**Comment by niboshi** [comment]
>>can I expect that the tests\chainer_tests\test_init_docstring.py issue will be solved in this PR?
>
>I don't have a plan to fix it.

It seems this error is only occurring in this PR.
I think I can't merge the PR unless it's fixed.

File: .pfnci/install_cupy.bat (hunk 2 above)
Comment (disktnk) [reply .pfnci/install_cupy.bat]: pushd / popd is good idea, thx

I think that installing cupy or other depend module on top repo directory is unsafe and should avoid.

**Comment by disktnk** [comment]
> It seems this error is only occurring in this PR.

I thought the error causes only Windows env.

I'd like to know how to stop the error on xpytest? CC: @imos @kmaehashi

**Comment by disktnk** [comment]
Ah, "`I thought the error causes only Windows env." it true, (misunderstanding is misunderstanding). I could reproduce the error without xpytest nor this PR, I reported an issue #7523

As you said, this PR cannot merge until the error is resolve, I'll wait it. @niboshi

File: .pfnci/run_test.bat [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    rem cannot set variable within if block, so use subroutine
+    cmd /c .pfnci\install_cupy.bat
+) else (
+    set PYTEST_ATTR=not gpu and not cudnn and %PYTEST_ATTR%
+)
+
+rem protobuf==3.7.1 with python3.7 causes "DeprecationWarning"
+rem to avoid the warning, install specific version
+rem see https://github.com/chainer/chainer/issues/7523
+pip install protobuf==3.6.1
```
Comment (toslunar) [inline .pfnci/run_test.bat]: The workaround could be removed because #7529 is merged and a newer `protobuf` is available.

**Comment by disktnk** [comment]
sorry for unattended, tests on windows os are failed, issued #7768. If need some time to fix the issue, CI can skip the target test.

**Comment by kmaehashi** [comment]
Blocked by https://github.com/chainer/chainer/pull/8465

**Comment by emcastillo** [comment]
#8465 was merged

**Pull request closed without merging**